### ISSUE TITLE
Adding a legend?

### ISSUE THREAD

**Issue by shriv** [developmentseed/lonboard#492]
The examples on `deck.gl` don't have an integrated legend but I saw [this suggestion on Pydeck](https://github.com/visgl/deck.gl/issues/4850#issuecomment-1188226561) and wonder if something similar would work with `lonboard`? There isn't an equivalent argument of `description` in the layers but it does look like [there is a description card component in `deck.gl` Jupyter widget](https://github.com/visgl/deck.gl/blob/02d6813584aa10d108e8cb28b7cc84ee6d6a31a9/modules/jupyter-widget/src/lib/components/description-card.js#L5) though not sure how to call it through `lonboard`..

**Comment by kylebarron** [comment]
Yes, a legend is desirable. But it's more likely for it to be built as a standalone widget (from python) than as a JavaScript component.

You can make a barebones legend by piecing together existing components from ipywidgets. A `vbox` of text and check boxes where checking a box toggles the `visible` flag on a layer

**Comment by shriv** [comment]
Hi @kylebarron. Just revisiting this issue as I do have pressing need for a colormap legend. I put together what I'm after.. The simplest option is putting the output of `cmap.mpl_colormap` on to the map. With Folium, you can use branca for legends. Can I take a similar approach? 

![image](https://github.com/user-attachments/assets/3bcc21c0-9f0f-4cc3-af0a-1b73126e7d99)

**Comment by kylebarron** [comment]
We don't currently have a way to inject custom HTML onto the map, and so this isn't currently possible. I don't think we want to provide a custom colormap legend _on the map itself_.

As it is, you can render the colormap in a cell right before the map. Or you can use something like `VBox` to render the colormap just above the map, but in the same cell output: https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Layout.html#the-vbox-and-hbox-helpers
